Nx tsconfig paths alias resolution is not working as expected

### Current Behavior

Hello !

The `@nx/vite/plugins/nx-tsconfig-paths.plugin` path resolution is not working as expected.

I think the first tsconfig is supposed to be the project tsconfig (https://github.com/nrwl/nx/blob/master/packages/vite/plugins/nx-tsconfig-paths.plugin.ts#L117) but in my case, it resolves to the tsconfig under the workspace root foder.


![Image](https://github.com/user-attachments/assets/396ab876-6595-4748-baf3-f1ffed8721e9)


### Expected Behavior

I think the first tsconfig is supposed to be the project tsconfig (https://github.com/nrwl/nx/blob/master/packages/vite/plugins/nx-tsconfig-paths.plugin.ts#L117) but in my case, it resolves to the tsconfig under the workspace root foder.

### GitHub Repo

https://github.com/nrwl/nx/blob/master/packages/vite/plugins/nx-tsconfig-paths.plugin.ts#L117

### Steps to Reproduce

1. root tsconfig

```json
{
  "compileOnSave": false,
  "compilerOptions": {
    "rootDir": ".",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "importHelpers": true,
    "target": "es2015",
    "module": "esnext",
    "lib": ["es2020", "dom"],
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "baseUrl": ".",
    "paths": {}
  },
  "exclude": ["node_modules", "tmp"]
}
```

2. project tsconfig

```json
{
  "compilerOptions": {
    "jsx": "react-jsx",
    "allowJs": true,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "types": ["vite/client", "vitest"],
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.app.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ],
  "extends": "../../tsconfig.base.json"
}

```


### Nx Report

```shell
Node           : 20.17.0
OS             : darwin-x64
Native Target  : x86_64-macos
npm            : 10.8.2

nx                 : 20.0.3
@nx/js             : 20.0.3
@nx/jest           : 20.0.3
@nx/eslint         : 20.0.3
@nx/workspace      : 20.0.3
@nx/cypress        : 20.0.3
@nx/devkit         : 20.0.3
@nx/eslint-plugin  : 20.0.6
@nx/react          : 20.0.3
@nx/vite           : 20.0.3
@nx/web            : 20.0.3
typescript         : 5.6.3
---------------------------------------
Registered Plugins:
@nx/vite/plugin
@nx/eslint/plugin
@nx/cypress/plugin
@nx/jest/plugin
---------------------------------------
```

### Failure Logs

```shell
[Nx Vite TsPaths] Resolved @/contexts to undefined
```

### Package Manager Version

_No response_

### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

Facing the same issue with executor @nx/jest:jest.. `nx run test` works fine from rootDir/frontend folder. But resolves modules incorrectly when run from the root folder. Always off by one `../` folder extra. Added module aliases in tsconfig.json, added moduleNameMapper to jest.config.js. No luck.

My frontend/jest.config.js file:
```
const path = require('path');

module.exports = {
  collectCoverage: true,
  collectCoverageFrom: [
    "2.0/**/**/*.{ts,tsx}",
    "!**/node_modules/**",
    "!**/vendor/**"
  ],
  coverageDirectory: 'coverage', // Output directory for coverage reports
  coverageReporters: ['html', 'text'], // Generate HTML and text coverage reports
  reporters: [
    'default'
  ],
  preset: 'react-native',
  transform: {
    '^.+\\.tsx?$': 'babel-jest',
  },
  rootDir: path.resolve(__dirname),
  roots: ['<rootDir>/2.0'], // Only look for tests inside the 2.0 folder
  testMatch: ['**/?(*.)+(spec|test).ts?(x)'],
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json'],
  moduleNameMapper: {
    '^~/(.*)$': '<rootDir>/src/$1',
    '^2\\.0/(.*)$': '<rootDir>/2.0/$1',
  }
};

console.log('Jest rootDir:', path.resolve(__dirname)); // prints `rootDir/frontend` as expected
```
My frontend/tsconfig.json
```
{
  "extends": "@react-native/typescript-config/tsconfig.json",
  "baseUrl": ".",
  "compilerOptions": {
    "noEmit": true,
    "paths": {
      "2.0/*": ["./2.0/*"],
      "~/*": ["./src/*"]
    }
  },
  "include": ["./src/**/*", "./2.0/**/*"]
}

```

Like mentioned by @Diboby it is incorrectly mapping modules with respect to root directory and not from my frontend folder. Unlike his issue, I don't even have a rootDir/tsconfig.json file.

When I run a spec in `2.0/components/typography/heading-text.spec.tsx`
It gives this error 
```
Cannot find module '../../../../2.0/core/utils' from '2.0/components/typography/heading-text.spec.tsx'
// when it should be looking instead for `../../../2.0/core/utils`
```

--- a/packages/vite/plugins/nx-tsconfig-paths.plugin.ts
+++ b/packages/vite/plugins/nx-tsconfig-paths.plugin.ts
@@ -20,6 +20,7 @@ import { Plugin } from 'vite';
 import { nxViteBuildCoordinationPlugin } from './nx-vite-build-coordination.plugin';
 import { findFile } from '../src/utils/nx-tsconfig-paths-find-file';
 import { isUsingTsSolutionSetup } from '@nx/js/src/utils/typescript/ts-solution-setup';
+import { getProjectTsConfigPath } from '../src/utils/options-utils';
 
 export interface nxViteTsPathsOptions {
   /**
@@ -210,17 +211,21 @@ export function nxViteTsPaths(options: nxViteTsPathsOptions = {}) {
   } as Plugin;
 
   function getTsConfig(preferredTsConfigPath: string): string {
+    const projectTsConfigPath = getProjectTsConfigPath(projectRoot);
     return [
       resolve(preferredTsConfigPath),
+      projectTsConfigPath ? resolve(projectTsConfigPath) : null,
       resolve(join(workspaceRoot, 'tsconfig.base.json')),
       resolve(join(workspaceRoot, 'tsconfig.json')),
       resolve(join(workspaceRoot, 'jsconfig.json')),
-    ].find((tsPath) => {
-      if (existsSync(tsPath)) {
-        logIt('Found tsconfig at', tsPath);
-        return tsPath;
-      }
-    });
+    ]
+      .filter(Boolean)
+      .find((tsPath) => {
+        if (existsSync(tsPath)) {
+          logIt('Found tsconfig at', tsPath);
+          return tsPath;
+        }
+      });
   }
 
   function logIt(...msg: any[]) {

